### PR TITLE
Clarified that QuerySet.filter() and Q() can both be empty.

### PR DESCRIPTION
**Pull request by adamchainz**

### CODE REVIEW

File: docs/ref/models/querysets.txt [hunk 1]
```diff
@@ -184,9 +184,9 @@ executed.
 Returns a new ``QuerySet`` containing objects that match the given lookup
 parameters.
 
-The lookup parameters (``**kwargs``) should be in the format described in
-`Field lookups`_ below. Multiple parameters are joined via ``AND`` in the
-underlying SQL statement.
+The zero or more lookup parameters (``**kwargs``) should be in the format
```
Comment (timgraham) [inline docs/ref/models/querysets.txt:187]: `**kwargs` are generally always optional. I think phrasing it this way does more to distract from common usage. (same statement could be written for exclude(), annotate(), etc.)

File: docs/topics/db/queries.txt [hunk 2]
```diff
@@ -1110,6 +1110,8 @@ that combine both a normal query and a negated (``NOT``) query::
 
     Q(question__startswith='Who') | ~Q(pub_date__year=2005)
 
+You may also create a ``Q`` with no arguments, which applies no lookups.
```
Comment (timgraham) [inline docs/topics/db/queries.txt:1113]: Outside the context of the mailing list thread (and in the context of these examples), this seems like a strange thing to do.

**Comment by felixxm** [comment]
@adamchainz Thanks for this patch :+1:, however I agree with Tim, I don't think it clarifies much about dynamic creation of combined (possibly empty) `Q` conditions.

**Comment by adamchainz** [comment]
Yeah fair enough. Rather than modifying the referenece it might be better having a worked example of building a Q dynamically.

**Pull request closed without merging**